src/infer_request.cc

```diff
@@ -1186,6 +1186,11 @@ InferenceRequest::Normalize()
       if (data_type == inference::DataType::TYPE_STRING) {
         RETURN_IF_ERROR(
             ValidateBytesInputs(input_id, input, &expected_byte_size));
+        // FIXME: -1 is used as a signal to skip total byte size validation for
+        // unhandled cases in ValidateBytesInputs.
+        if (expected_byte_size == -1) {
+          return Status::Success;
```

Comment:
I don't think we should return immediately incase more checks are appended to this function.

Reply:
Open to changes on the GPU portions if you've already been looking into it. The early return can be changed if new checks are added or if we add support for checking the GPU buffers. 

The early return here is temporary to avoid failing the total element count validation at the end if we skip elements in gpu buffers.

Reply:
My concern is someone started appending new sections without looking backwards. You can change to
```
bool skip_byte_size_check = (expected_byte_size == -1);

if (!skip_byte_size_check &&
    ((byte_size > INT_MAX) || (static_cast<int64_t>(byte_size) != expected_byte_size))) {
```